code/modules/surgery/organs/internal/cyberimp/augments_internal.dm

```diff
@@ -127,11 +131,25 @@
 		addtimer(CALLBACK(src, PROC_REF(clear_stuns)), stun_cap_amount, TIMER_UNIQUE|TIMER_OVERRIDE)
 
 /obj/item/organ/internal/cyberimp/brain/anti_stun/proc/clear_stuns()
-	if(owner || !(organ_flags & ORGAN_FAILING))
+	if(owner && !(organ_flags & ORGAN_FAILING) && (COOLDOWN_FINISHED(src, implant_cooldown)))
 		owner.SetStun(0)
 		owner.SetKnockdown(0)
 		owner.SetImmobilized(0)
 		owner.SetParalyzed(0)
+		owner.setStaminaLoss(0)
+		ADD_TRAIT(owner, TRAIT_IGNOREDAMAGESLOWDOWN, REF(src))
+		addtimer(TRAIT_CALLBACK_REMOVE(owner, TRAIT_IGNOREDAMAGESLOWDOWN, REF(src)), stun_resistance_time)
+		ADD_TRAIT(owner, TRAIT_BATON_RESISTANCE, REF(src))
+		addtimer(TRAIT_CALLBACK_REMOVE(owner, TRAIT_BATON_RESISTANCE, REF(src)), stun_resistance_time)
+		ADD_TRAIT(owner, TRAIT_STUNIMMUNE, REF(src))
+		addtimer(TRAIT_CALLBACK_REMOVE(owner, TRAIT_STUNIMMUNE, REF(src)), stun_resistance_time)
+		addtimer(CALLBACK(owner, TYPE_PROC_REF(/mob/living, setStaminaLoss), 0), stun_resistance_time)
+		COOLDOWN_START(src, implant_cooldown, 60 SECONDS)
+		addtimer(CALLBACK(src, PROC_REF(implant_ready)),60 SECONDS)
```

Comment:
turn this into an early return there's no reason for there to be so many lines on the indent here

```dm
if(isnull(owner) || (organ_flags & ORGAN_FAILING) || !COOLDOWN_FINISHED(src, implant_cooldown))
 return // this should be a tab instead of a space before your compiler fails because i can't copypaste one here

// rest of the code on this indentation
```

Reply:
also use `add_traits()` like `owner.add_traits(list(TRAIT_IGNOREDAMAGESLOWDOWN, TRAIT_BATON_RESISTANCE, TRAIT_STUNIMMUNE), REF(src))` and then keep all the `addtimer()` calls together, the mixing and matching isn't fun to read.

Reply:
Done; it should look cleaner now.